pkg/controller/compliancescan/scan.go

```diff
@@ -645,16 +645,21 @@ func checkScanUnknownError(cm *corev1.ConfigMap) error {
 	return nil
 }
 
-func getScanResult(cm *corev1.ConfigMap) (compv1alpha1.ComplianceScanStatusResult, error) {
+func scanResultReady(cm *corev1.ConfigMap) bool {
 	if cm.Annotations == nil {
-		return compv1alpha1.ResultError, fmt.Errorf("the ConfigMap '%s' was missing annotations", cm.Name)
+		return false
 	}
 
-	strResult, ok := cm.Annotations[compv1alpha1.CmScanResultAnnotation]
+	_, ok := cm.Annotations[compv1alpha1.CmScanResultAnnotation]
```

Comment:
If this won't be an error anymore, perhaps we want to log, at least.

In the case where a user would manually modify a ConfigMap, and remove the annotation, the scan would never get to ready state.

Reply:
I've actually reverted to the old behaviour. We now have a guard before we call `getScanResult` function to check if the CM is ready, so calling the `getScanResult` function with invalid input should throw an error as you say.

Reply:
Hmm, wait up, I know what you meant, let me add one more logging call.

Reply:
Actually, the call to `scanResultReady` already logs something, hope it's sufficient:
``+                       cmHasResult := scanResultReady(foundCM)                                                                                                                                                                              
+                       if cmHasResult == false {                                                                                                                                                                                            
+                               logger.Info("Node ConfigMap is not ready yet", "node.Name", node.Name)                                                                                                                                       
+                               isReady = false                                                                                                                                                                                              
+                               continue                                                                                                                                                                                                     
+                       }                                                                                                                                                                                                                    
+                         `

```

Reply:
Right, it already looks that the node is not ready. But you don't know why. And we have no docs about it or relevant logs.

Reply:
A new log message was added per suggestion on Slack.